### DISCUSSION TITLE
How to make all labels the same size as the largest label?

### DISCUSSION THREAD

**Discussion by WarWithinMe** [AvaloniaUI/Avalonia#12853] [Q&A] [answered]
Say I have 10 labels, each one has different text, so each one has a different size. I want all the labels to be as wide as the widest label.

The labels might be in different view hierarchies so it might not be possible to use Panels or DataGrid to layout the labels.

Previously, in axaml, I would make a control's width bound to another control's width. But here I don't know which one is the widest until layouting is done.

Should I implement this in code behind (and if it is, when should I gather the labels' information and update the labels)? Or is there a clever way to do it via axaml and binding?

**Comment by timunie** [comment, marked as answer]
You can use 10 Grids with SharedGridSize to achieve this.

**Reply by WarWithinMe** [reply]
Thanks for the answer. I assume it's `SharedSizeGroup`. For reference, [here](https://github.com/AvaloniaUI/Avalonia/issues/10907#issuecomment-1495888065) is an example for how to use it. Notice that one must set `Grid.IsSharedSizeScope="True"` on a common ancestor.

**Reply by timunie** [reply]
Exactly 👍
